test/test_jit_string.py

```diff
@@ -1,5 +1,6 @@
 from test_jit import JitTestCase
-
+import sys
+sys.setrecursionlimit(10000)
```

Comment:
Why is this needed?

Reply:
it won't pass the tests on my machine without it -- the default seems to conservative

Reply:
what is causing the big recursion? Is it not suspicious?